programs/psitest/main.cpp

```diff
@@ -465,12 +525,27 @@ struct callbacks
       throw std::runtime_error("called testerAbort");
    }
 
-   void testerExit(int32_t)
+   void testerWasi_proc_exit(int32_t code)
    {
       backtrace();
       throw std::runtime_error("called testerExit");
    }
 
+   int32_t testerWasi_sched_yield()
+   {
+      return 0;
+   }
+
+   // Unfortunately it looks like we have to lie instead of returning an error
+   // to not break some Rust libraries and even standard ones like HashMap
+   uint32_t testerWasi_random_get(wasm_ptr<uint8_t> buf, wasi_size_t buf_len)
+   {
+      if (buf_len > 0) {
+         *buf = 0;
```

Comment:
I got it from this original rust implementation located here:
```rs
// Unfortunately it looks like we have to lie instead of returning an error
// to not break some Rust libraries
#[no_mangle]
pub unsafe extern "C" fn random_get(mut buf: *mut u8, mut buf_len: Size) -> Errno {
    while buf_len > 0 {
        *buf = 0;
        buf = buf.offset(1);
        buf_len -= 1;
    }
    0
}
```

I guess by returning only zeros is better to ensure determinism? Thoughts?

BTW, I realized this is what is polyfilled for rust services: https://github.com/gofractally/psibase/blob/main/rust/cargo-psibase/service_wasi_polyfill/src/lib.rs#L39-L49. Do you guys anticipate any deterministic problems happening here? 
I know it is used at least for rust HashMaps. 

cc @James-Mart

Reply:
There are two types of problems that are likely to happen if the result is a fixed value:
- If the random numbers are being used for security purposes, then making them predictable defeats the point. I don't think we actually care about security for unit tests, so this isn't very important.
- Certain algorithms that manipulate random variates can go into an infinite loop if the random number generator always returns the same value.

Reply:
Specifically for the Rust Hashmaps, I think we are okay. My understanding is that randomness is used because of the choice of the default hash algorithm, which is used because it is resistant to HashDos attacks. Breaking the randomness makes the data structure vulnerable to HashDos attacks, which is probably okay in our case because compute is billed.

Polyfilling it to be deterministic was what Todd originally implemented. It does feel a bit suspicious to me, like we don't know what else we might be breaking in libraries that services use that are supposed to depend on randomness. @swatanabe what do you think?